sdk/analytics/src/main/java/com/klaviyo/analytics/model/Event.kt

```diff
@@ -17,9 +17,9 @@ class Event(val type: EventType, properties: Map<EventKey, Serializable>?) :
 
     constructor(type: String) : this(type, null)
 
-    fun setValue(value: String) = apply { this.value = value }
-    var value: String
-        get() = (this[EventKey.VALUE] ?: "") as String
+    fun setValue(value: Double) = apply { this.value = value }
```

Comment:
should we allow setting null (clearing the value)?

Reply:
tru

Reply:
after looking at the profile model, should those setter methods also allow null?

Reply:
Agreed. Also thinking we might need type validation for those keys in the `setProperty` method so you can't work around the typing that way.

Reply:
🎶 edge cases 🎶

Reply:
could maybe use `.toString` instead of casting in some cases, seems more graceful or yeah we can enforce types more strictly

Reply:
yeah, it already has to be serializable, so toString is probably sufficient and simpler

Reply:
looks like you missed updating this method to accept null

Reply:
certainly did. fixed now!